iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothubservices/FileUploadTests.java

```diff
@@ -578,7 +643,185 @@ public void run()
+        if (!executor.awaitTermination(10000, TimeUnit.MILLISECONDS))
+        {
+            executor.shutdownNow();
+        }
+
+        for (int i = 1; i < MAX_FILES_TO_UPLOAD; i++)
+        {
+            assertEquals("File" + i + " has no notification", fileUploadState[i].fileUploadNotificationReceived, SUCCESS);
+        }
+
+        tearDownDeviceClient();
+    }
+
+    @Test (timeout = MAX_MILLISECS_TIMEOUT_KILL_TEST)
+    public void uploadToBlobAsyncAndTelemetryOnMQTTSelfSigned() throws URISyntaxException, IOException, InterruptedException, ExecutionException, TimeoutException
+    {
+        // arrange
+        setUpX509DeviceClient(IotHubClientProtocol.MQTT);
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+
+        // act
+        for (int i = 1; i < MAX_FILES_TO_UPLOAD; i++)
+        {
+            final int index = i;
+            executor.submit(new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    try
+                    {
+                        deviceClient.uploadToBlobAsync(fileUploadState[index].blobName, fileUploadState[index].fileInputStream, fileUploadState[index].fileLength, new FileUploadCallback(), fileUploadState[index]);
+                    }
+                    catch (IOException e)
+                    {
+                        fail(e.getMessage());
+                    }
+                }
+            });
+
+            executor.submit(new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    deviceClient.sendEventAsync(new com.microsoft.azure.sdk.iot.device.Message(messageStates[index].messageBody), new FileUploadCallback(), messageStates[index]);
+                }
+            });
+
+            FileUploadNotification fileUploadNotification = getFileUploadNotificationForThisDevice(scDevicex509);
+
+            // assert
+            verifyNotification(fileUploadNotification, fileUploadState[i]);
+
+            assertTrue(fileUploadState[i].isCallBackTriggered);
+            assertEquals(fileUploadState[i].fileUploadStatus, SUCCESS);
+            assertEquals(messageStates[i].messageStatus, SUCCESS);
+        }
+
+        executor.shutdown();
+        if (!executor.awaitTermination(10000, TimeUnit.MILLISECONDS))
+        {
+            executor.shutdownNow();
+        }
+
+        for (int i = 1; i < MAX_FILES_TO_UPLOAD; i++)
+        {
+            assertEquals("File" + i + " has no notification", fileUploadState[i].fileUploadNotificationReceived, SUCCESS);
+        }
+
+        tearDownDeviceClient();
+    }
+
+    @Test (timeout = MAX_MILLISECS_TIMEOUT_KILL_TEST)
+    public void uploadToBlobAsyncSingleFileOnAMQPSelfSigned() throws URISyntaxException, IOException, InterruptedException, ExecutionException, TimeoutException
+    {
+        // arrange
+        setUpX509DeviceClient(IotHubClientProtocol.AMQPS);
```

Comment:
WS?

Reply:
found it up there....nevermind

---
In reply to: [237619247](https://github.com/Azure/azure-iot-sdk-java/pull/384#discussion_r237619247) [](ancestors = 237619247)